frontend/src/metabase/admin/performance/types.tsx

```diff
@@ -0,0 +1,146 @@
+import { c, t } from "ttag";
+import type { AnySchema } from "yup";
+
+import {
+  strategyValidationSchema,
+  doNotCacheStrategyValidationSchema,
+  ttlStrategyValidationSchema,
+  durationStrategyValidationSchema,
+  inheritStrategyValidationSchema,
+} from "./validation";
+
+type StrategyData = {
+  label: string;
+  shortLabel?: string;
+  validateWith: AnySchema;
+};
+
+export type StrategyType = "nocache" | "ttl" | "duration" | "inherit";
+
+/** Cache invalidation strategies and related metadata */
+export const Strategies: Record<StrategyType, StrategyData> = {
+  ttl: {
+    label: t`TTL: When the time-to-live (TTL) expires`,
+    shortLabel: c("'TTL' is short for 'time-to-live'").t`TTL`,
+    validateWith: ttlStrategyValidationSchema,
+  },
+  duration: {
+    label: t`Duration: after a specific number of hours`,
+    validateWith: durationStrategyValidationSchema,
+    shortLabel: t`Duration`,
+  },
+  nocache: {
+    label: t`Don't cache results`,
+    validateWith: doNotCacheStrategyValidationSchema,
+    shortLabel: t`No caching`,
+  },
+  inherit: {
+    label: t`Use default`,
+    validateWith: inheritStrategyValidationSchema,
+  },
+};
+
+export const getStrategyLabel = (strategy?: Strategy) => {
+  return strategy ? Strategies[strategy.type].label : null;
+};
+
+export const getShortStrategyLabel = (strategy?: Strategy) => {
+  if (!strategy) {
+    return null;
+  }
+  const type = Strategies[strategy.type];
+  return type.shortLabel ?? type.label;
+};
+
+export const isValidStrategyName = (
+  strategy: string,
+): strategy is StrategyType => {
+  return Object.keys(Strategies).includes(strategy);
```

Comment:
prefer Set created outside of this function (avoid allocating a new array on every check) and use the O(1) `.has` method